packages/sqs/lib/sqs/AbstractSqsPublisher.ts

```diff
@@ -64,15 +72,12 @@ export abstract class AbstractSqsPublisher<MessagePayloadType extends object>
       }
 
       message = this.updateInternalProperties(message)
+      const maybeOffloadedPayloadMessage = await this.offloadMessagePayload(
+        message,
+        calculateOutgoingMessageSize(message),
```

Comment:
can we do the calculation inside, after we have already passed the check of `is offloading enabled`? otherwise we are making the flow slower for everyone, including those who don't use the functionality

Reply:
Fair point.
I was trying to avoid requiring calculateOutgoingMessageSize() in the abstraction and then stubbing it in the Consumer implementation.
What if we pass the size calculation as a callable that will be invoked when needed?
![Screenshot 2024-06-17 at 15 22 58](https://github.com/kibertoad/message-queue-toolkit/assets/7249095/0e480d4e-11dd-4025-96db-e2a22794afe0)

Reply:
yup, sounds good!